### ISSUE TITLE
now-cli should not start the deployment until it got an answer from the "public" confirm question

### ISSUE THREAD

**Issue by idealhack** [vercel/vercel#1020]
After #1019 

I add the `EXPOSE` line and enter an `now` command - it’s deploying when I haven’t answer the `You are on the OSS plan. Your code and logs will be made public.` question, and it’s done after a few secs, the app is also running!

Then I entered `^C`, it says `Deployment aborted. No files were synced.` Now the deployment were deleted - but the app haven’t been killed and I have no way to stop it.

I think `now` should not start the deployment until it got an answer of that question. Below is the output:

```
$ now
> > Deploying ~/go/src/github.com/example/app under mail@example.com
> Ready! https://example.now.sh (copied to clipboard) [4s]
> You (mail@example.com) are on the OSS plan. Your code and logs will be made public.
> Deployment aborted. No files were synced.
  You can upgrade by running `now upgrade`.
```

P.S. This is an issue I posted on slack - https://zeit-community.slack.com/archives/C1TMPLCKS/p1512383080000111

**Comment by leo** [comment]
We did this on purpose, so that the deployment already gets up and running before you even decide to answer. Otherwise all of those things will be blocked by the prompt.

And if you're away from the terminal or even when focused, this makes the product seem much slower than it could actually be - at no price.

Because in my opinion, even if the logs and code are public, nobody will ever access them, since you haven't yet shared the URL. 😊

**Comment by leo** [comment]
But we should definitely rephrase it to indicate that we've already started uploading and that answering with "no" will abort it.

**Comment by idealhack** [comment]
I can understand this purpose and I thought it's acceptable. It would be nicer if you tells the users :)

But the problem is this will leads to a bug as I said. Aborting deleted the deployment but did not kill the app.

I have not try to reproduce this in a normal workflow.

**Comment by leo** [comment]
@idealhack Yes, aborting the deployment never kills it. However, this is also intentional - but other people already reported this is a bug as well, so let's talk about this in https://github.com/zeit/now-cli/issues/752.

In the meantime, we'll definitely rephrase the question! 😊 

I also want to not that - if you choose "No" when the prompt pops up, the deployment will definitely be removed.

**Comment by idealhack** [comment]
I'm sorry but I can't see issue 752 related to the killing-app issue...

I wonder are there any GC policies here? This may be OK to a website as there're no state basically, but if it's an realtime app connecting to an API, this will cause conflict and block newer versions from connecting the the same API.

**Comment by idealhack** [comment]
Seems my app from an removed deployment got killed somehow after a few hours. So are there any documents of this? I still don't know if an app will got killed when a deployment been removed.

**Comment by idealhack** [comment]
I’m closing this because it seems to be inactive for months. If anyone feel this issue still matters, feel free to reopen it.
